drivers/i3c/i3c_cdns.c

```diff
@@ -1732,6 +1739,8 @@ static int cdns_i3c_transfer(const struct device *dev, struct i3c_device_desc *t
 		} else {
 			send_broadcast = true;
 		}
+
+		cmd->num_xfer = msgs[i].num_xfer;
```

Comment:
Shouldn't this be the other way around? `msgs[i].num_xfer = cmd->num_xfer`?

Reply:
no, this is just writing the pointer for reference, which is then updated within `cdns_i3c_complete_transfer`